### PR TITLE
Fix validation timeout

### PR DESCRIPTION
**Pull request by l-vincent-l**
Replace the validation button by a live view.
We can now track what is happening, and we don't have timeout anymore.
The important part is here https://github.com/etalab/transport-site/compare/master...l-vincent-l:fix_validation_timeout?expand=1#diff-093e014b4accb0c3cabcf08d9193903e

### CODE REVIEW

File: apps/transport/lib/transport_web/api/schemas.ex [hunk 1]
```diff
@@ -20,13 +20,19 @@ defmodule TransportWeb.API.Schemas do
...
           enum: ["Point", "LineString", "Polygon", "MultiPoint", "MultiLineString", "MultiPolygon"]
         }
-      },
-      discriminator: %Discriminator{
-        propertyName: "type"
       }
     }
   end
 
+  defmodule NumberItems do
```
Comment (antoine-de) [inline apps/transport/lib/transport_web/api/schemas.ex]: by curiosity why did you change this?

File: .credo.exs [hunk 2]
```diff
@@ -22,7 +22,7 @@
         # In the latter case `**/*.{ex,exs}` will be used.
       #
         included: ["lib/", "src/", "web/", "apps/"],
-        excluded: [~r"/_build/", ~r"/deps/"]
+        excluded: [~r"/_build/", ~r"/deps/", ~r"/apps/transport/client/"]
```
[the remark this answers is not in the document]
Comment (l-vincent-l) [reply .credo.exs]: Because it's suppose to be JS and live view writes some elixir in it that doesn't respect our credo conf.

File: apps/gbfs/mix.exs [hunk 3]
```diff
@@ -33,6 +33,7 @@ defmodule GBFS.MixProject do
       {:phoenix, "~> 1.4"},
       {:iconv, "~> 1.0.10"},
       {:sweet_xml, ">= 0.0.0"},
+      {:poison, ">= 0.0.0"}
```
[the remark this answers is not in the document]
Comment (l-vincent-l) [reply apps/gbfs/mix.exs]: I think so.
But I did that because in the other repo we already have a dependency to it (via ecto), so I don't want to fix a version here.

File: apps/transport/lib/transport_web/api/schemas.ex (hunk 1 above)
Comment (l-vincent-l) [reply apps/transport/lib/transport_web/api/schemas.ex]: Because it was compatible with the new version of openapispex

**Comment by l-vincent-l** [comment]
> tada nice!
> 
> OK for me apart from a small display glitch now that the size of the text can change:
> image
> 
> Also for the "validated" phase there seems to also be a display problem:
> image
> 
> Also to be precise I don't think this correct the timeout bug (I still sometimes get it for IDFM), but this brings feedback on the validating process and it's very nice!

I tried to fix the glitch, but I would need fix row sizes to do that, and it's not really covenient...
So for now I've done something live that looks better

image
image
